packages/compiler/core/types.ts

```diff
@@ -41,6 +41,15 @@ export interface BaseType {
   projectionSource?: Type;
   projectionBase?: Type;
   projector?: Projector;
+
+  /**
+   * Reflect if a type has been finished(Decorators have been called).
+   * There is multiple reasons a type might not be finished:
+   * - a template declaration will not
```

Comment:
Will not be finished?

Reply:
yeah finished being when decorators are run. Decorators are not run on a template declaration